### DISCUSSION TITLE
How to use Prisma $transaction api in this scenario?

### DISCUSSION THREAD

**Discussion by s524797336** [prisma/prisma#6355] [Q&A]
I found that there is a problem with my previous question on stackoverflow, but I think this is also a commonly used coding style, so does anyone think my reproduce is a real world problem?

Question at stackoverflow:
https://stackoverflow.com/questions/66863807/how-to-use-prisma-transaction-api-in-this-scenario

Small reproduce:
https://github.com/s524797336/test-prisma-repo-for-Ryan

**Comment by ryands17** [comment]
@s524797336 👋 

In your example, you have passed an async function to `prisma.$transaction`. This is not supported and it only accepts a `PrismaPromise`. So you will face this error:

![image](https://user-images.githubusercontent.com/19697099/113698872-873b6200-96f2-11eb-8f38-db0b6b518a04.png)

**Reply by ryands17** [reply]
If you run the transactions in the following manner as per the reproduction that you have sent me, then the first record will not be created:

```ts
function createA() {
    return prisma.test.create({ data: { name: 'a' } })
}

function createB() {
    const b = prisma.test.create({ data: { name: 'b' } })
    return b
}

async function main() {
    const a = createA()
    throw new Error()
    const b = createB()
    await prisma.$transaction([a, b])
}
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
